Frameworks/UIKit/UIView.mm

```diff
@@ -197,6 +222,40 @@ - (bool)_processGesturesForTouch:(UITouch*)touch event:(UIEvent*)event touchEven
 
             if (send) {
                 [curgesture performSelector:eventName withObject:[NSMutableSet setWithObject:touch] withObject:event];
+                // verify if gesture state is Began or recognized after transition
+                if ((curgesture.state == UIGestureRecognizerStateBegan) || (curgesture.state == UIGestureRecognizerStateRecognized)) {
+                    gestureOnGoing = YES;
+                    if (DEBUG_GESTURES) {
+                        TraceVerbose(TAG, L"gesture %hs is in state %d, cancel DMGesture.", object_getClassName(curgesture), curgesture.state);
+                    }
+                }
+            }
+        }
+    }
+
+    // scanning DM Getures, if a gestures is on going, cancel DM Gestures
+    // otherwise, send Touch to DM gestures
+    for (int i = 0; i < dmGestureCount; i++) {
+        UIGestureRecognizer* dmGesture = dmrecognizers[i];
+        if (gestureOnGoing) {
+            [dmGesture _cancelIfActive];
+            if (DEBUG_GESTURES) {
+                TraceVerbose(TAG, L"Cancelled DMGesture %hs .", object_getClassName(dmGesture));
+            }
+        } else {
+            if ([dmGesture state] != UIGestureRecognizerStateCancelled) {
+                id delegate = [dmGesture delegate];
+                BOOL send = TRUE;
+                if (touch.phase == UITouchPhaseBegan && [delegate respondsToSelector:@selector(gestureRecognizer:shouldReceiveTouch:)]) {
+                    send = [delegate gestureRecognizer:dmGesture shouldReceiveTouch:touch];
+                }
+
+                if (send) {
+                    [dmGesture performSelector:eventName withObject:[NSMutableSet setWithObject:touch] withObject:event];
+                    if (DEBUG_GESTURES) {
+                        TraceVerbose(TAG, L"Send Touch with phase=%d to DMGesture %hs.", touch.phase, object_getClassName(dmGesture));
+                    }
```

Comment:
the current form is actually better. notice default is true. If remove send, the conditional clause becomes much more complex and not clear. 

---
In reply to: [86438414](https://github.com/Microsoft/WinObjC/pull/1305#discussion_r86438414) [](ancestors = 86438414)